packages/shorebird_cli/test/src/commands/preview_command_test.dart

```diff
@@ -1267,6 +1278,376 @@ channel: ${DeploymentTrack.staging.channel}
+            ),
+          ).thenAnswer((_) async {});
+          when(
+            () => bundletool.installApks(
+              apks: any(named: 'apks'),
+              deviceId: any(named: 'deviceId'),
+            ),
+          ).thenAnswer((_) async {});
+          when(
+            () => adb.clearAppData(
+              package: any(named: 'package'),
+              deviceId: any(named: 'deviceId'),
+            ),
+          ).thenAnswer((_) async {});
+          when(
+            () => adb.startApp(
+              package: any(named: 'package'),
+              deviceId: any(named: 'deviceId'),
+            ),
+          ).thenAnswer((_) async {});
+          when(
+            () => adb.logcat(
+              filter: any(named: 'filter'),
+              deviceId: any(named: 'deviceId'),
+            ),
+          ).thenAnswer((_) async => process);
+          when(() => process.stdout).thenAnswer((_) => const Stream.empty());
+          when(() => process.stderr).thenAnswer((_) => const Stream.empty());
+          when(() => releaseArtifact.url).thenReturn(androidReleaseArtifactUrl);
+          when(
+            () => process.exitCode,
+          ).thenAnswer((_) async => ExitCode.success.code);
+
+          createAabFile(channel: null);
+        });
+
+        test('exits with success and calls install and launch app', () async {
+          final exitCode = await runWithOverrides(command.run);
+          expect(exitCode, equals(ExitCode.success.code));
+
+          verify(
+            () => bundletool.installApks(
+              apks: any(named: 'apks'),
+              deviceId: deviceId,
+            ),
+          ).called(1);
+        });
+
+        group('when the ios release is not sideloadable', () {
+          setUp(() {
+            final unfilteredRelease = MockRelease();
+            when(() => unfilteredRelease.id).thenReturn(releaseId);
+            when(() => unfilteredRelease.version).thenReturn(releaseVersion);
+            when(() => unfilteredRelease.platformStatuses).thenReturn({
+              ReleasePlatform.ios: ReleaseStatus.active,
+              ReleasePlatform.android: ReleaseStatus.active,
+            });
+            when(() => release.platformStatuses).thenReturn({
+              ReleasePlatform.android: ReleaseStatus.active,
+            });
+            when(
+              () => codePushClientWrapper.getReleases(
+                appId: any(named: 'appId'),
+                sideloadableOnly: true,
+              ),
+            ).thenAnswer((_) async => [release]);
+            when(
+              () => codePushClientWrapper.getReleases(
+                appId: any(named: 'appId'),
+              ),
+            ).thenAnswer((_) async => [unfilteredRelease]);
+          });
+
+          test('warns about the platform and goes directly to iOS', () async {
+            final exitCode = await runWithOverrides(command.run);
+            expect(exitCode, equals(ExitCode.success.code));
```

Comment:
I don't know that we need to warn about iOS if the user specified that they want to preview Android.

Reply:
Also, should the test name be "goes directly to android"?

Reply:
ah, this message is totally wrong, it was a remain from early tests that I was doing and we changed the approach since then.

I rewrote the message